omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_bootstrap_validator.rb

```diff
@@ -92,14 +92,8 @@ def pivotal_key_exists?
   end
 
   def validate_sane_state
-    if pivotal_key_exists?
-      unless secrets_exists?
+    if pivotal_key_exists? && !secrets_exists?
         fail_with err_BOOT006_pivotal_with_no_secrets
-      end
-    else
-      if secrets_exists?
-        fail_with err_BOOT007_secrets_with_no_pivotal
```

Comment:
> it seems it is possible (through Ctrl+C) to end up in a situation where the secrets file exists containing no secrets other than the pivotal key, because we create it in the 'compile' phase of the chef client run.

This is true, but the next (successful) configure run would fix this. And it would be odd to have the preflight check _fail_ because of something fixable, wouldn't it?

> Continuing the thought, the question there is.. is it an error? I think it still can be, but it's a condition we might force to occur in the "ctrl+c to do something external and then run reconfigure again" instruction for HA.

I'm not sure I follow, would you want to _keep_ BOOT007? I'd assume we could blow this up, to have preflight checks around the existence of secrets in the store, but since stuff would be put there _later_, I'm not sure that there's a valid thing we could catch with it.